src/coord/src/coord.rs

```diff
@@ -3301,8 +3301,24 @@ impl Coordinator {
         }
     }
 
-    fn set_index_options(&mut self, id: GlobalId, options: Vec<IndexOption>) {
-        let index = self.indexes.get_mut(&id).expect("index known to exist");
+    fn set_index_options(
+        &mut self,
+        id: GlobalId,
+        options: Vec<IndexOption>,
+    ) -> Result<(), CoordError> {
+        let index = match self.indexes.get_mut(&id) {
+            Some(index) => index,
+            None => {
+                if !self.catalog.is_index_enabled(&id) {
```

Comment:
Why is it important to prevent users from changing index options while disabled? Seems to me like users might benefit from being able to adjust some of these parameters while disabled then restart/enable and see how they perform.